### ISSUE TITLE
addAggregateFunction and isValid

### ISSUE THREAD

**Issue by cristipp** [uwdata/arquero#196]
I need to implement the aggregate function `op.every(col)` such that it returns `false` if at least one value is `false` or `undefined`, and returns `true` otherwise. 

A naive attempt using addAggregateFunction does not work because `null` / `undefined` values are filtered out before `add` is invoked by an `if (isValid(value))` check. See https://github.com/uwdata/arquero/blob/fce414576794cbd8d48089116e554404b79ef9a7/src/engine/reduce/field-reducer.js#L120.

In dplyr ecosystem this is customizable via the `na.rm = ...` parameter. Note that by default `na.rm` *is* `FALSE`, keeping all values. See https://dplyr.tidyverse.org/reference/summarise_all.html#examples

This is fixable by adding a check when retrieving the value, though 'state.count' and 'state.valid' are undocumented:

```
    value: (state) => {
        return state.count === state.valid && state.every
    }
```

Are there plans to support something along the lines of `na.rm = ...`, and/or document state.count / state.valid variables?

**Comment by jheer** [comment]
Thanks for the issue, and glad you successfully added the `every` aggregate! There are no plans at this point to support something akin to `na.rm`, but I will track this over time to see if there is a clear need.

We will update the extensibility documents to provide more details about the aggregate state properties.

**Comment by jheer** [comment]
Updated documentation will be included as part of the 4.8.0 release. In the meantime, I've also published this notebook on adding aggregate functions, partly inspired by this issue (and which includes an alternate implementation of `op.every()`: https://observablehq.com/@uwdata/adding-aggregate-functions-to-arquero
